integration-tests/sanity-functional-protractor/specs/api-spec.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+//npm test -- --specs specs/api-spec.js
+
+const PublishPage = require('../page_objects/abstract-po.js');
+const AetsApi = require('../page_objects/api-po.js');
+const testDataJson = require('../data/test_data/api-data.json');
+const chakram = require('chakram');
+const chakramExpect = chakram.expect;
+const jasmineExpect = expect;
+
+describe('AET reports web application - Rest API tests', () => {
+
+    const HTTP_200_OK = 200;
+    const HTTP_404_NOT_FOUND = 404;
+    const HTTP_400_BAD_REQUEST = 400;
+
+    describe('Get metadata by suite name api - validation', () => {
+
+        it('JSON Schema validation', () => {
+            const parametersSuiteName = testDataJson.get_metadata_by_suite_name.parameters;
+            const jsonSchemaPath = require(testDataJson.get_metadata_by_suite_name.json_schema_path);
+            return AetsApi.getMetadata(parametersSuiteName).then((response) => {
+                chakramExpect(response).to.have.status(HTTP_200_OK);
+                chakramExpect(response).to.have.schema(jsonSchemaPath);
+            });
+        });
+
+        it('Error message validation', () => {
+            const parametersSuiteName = testDataJson.get_metadata_by_suite_name.parameters;
+            const parametersNoExistingSuiteName = clone(parametersSuiteName);
+            parametersNoExistingSuiteName.suite = "no_existing_suite"
+            const errorMessageJson = testDataJson.get_metadata_by_suite_name.error_message;
+            return AetsApi.getMetadata(parametersNoExistingSuiteName).then((response) => {
+                chakramExpect(response).to.have.status(HTTP_400_BAD_REQUEST);
+                jasmineExpect(response.body).toEqual(errorMessageJson);
+            });
+        });
+    });
+    describe('Get metadata by correlationId - validation', () => {
+
+        it('JSON Schema validation', () => {
+            const parametersSuiteName = testDataJson.get_metadata_by_suite_name.parameters;
+            const parametersCorrelationId = testDataJson.get_metadata_by_correlationId.parameters;
+            const jsonSchemaPath = require(testDataJson.get_metadata_by_correlationId.json_schema_path);
```

Comment:
Wouldn't be more clean to declare all of the requires at the beginning of the file?

Reply:
I've decreased number of declaration and moved then to the beginning